### PR TITLE
plugins/dap: ensure extensions load after the main plugin

### PR DESCRIPTION
**Pull request by Wolbyte**
This pull request fixes issues with `nvim-dap` and extensions that add custom configurations/adapters to `nvim-dap`. All extensions now load after `nvim-dap` is initialized to prevent user-defined settings from overriding the configurations added by them.

### CODE REVIEW

File: plugins/dap/default.nix
Comment (GaetanLepage) [inline plugins/dap/default.nix:48]:
```suggestion
        extensionConfigLua = mkOption {
          type = types.lines;
          description = ''
            Extension configuration for dap. Don't use this directly !
          '';
          default = "";
          internal = true;
        };
```

I think that this would fit better for this option.

File: plugins/dap/default.nix
Comment (GaetanLepage) [inline plugins/dap/default.nix:92]:
```suggestion
          + cfg.extensionConfigLua);
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Very nice solution !
Thanks for the fix :)

**Review by GaetanLepage — APPROVED** [review]
LGTM !
I prefer to wait for @traxys 's review before merging.

**Review by traxys — APPROVED** [review]

**Pull request merged**
